### ISSUE TITLE
Label transfer crashes when too many cells are there

### ISSUE THREAD

**Issue by dawe** [dawe/schist#60]
Apparently label transfer, actually the step when affinities are calculated, can crash due to memory error when too many cells are there, and "many" is not even high (>25k). 
This probably is due to the fact schist tries to address a numpy array which is too big to be managed. Going with sparse implementations could be the way, but I'm afraid the affinity matrix is not really sparse, unless we set a threshold under which every thing is actually 0

**Comment by dawe** [comment]
Actually it's on `graph-tool`, as it crashes when tries to initialize a blockstate

```
> /beegfs/scratch/ric.cosr/cittaro.davide/src/schist/schist/tools/_affinity_tools.py(119)calculate_affinity()
-> state = gt.BlockState(g, b=partitions)
(Pdb) 
Segmentation fault (core dumped)
```

**Comment by dawe** [comment]
Ok, I've figured this out. `graph-tool` crashes when partitions have negative values, this happens when there are `nan` in the label to be assigned. It should be easy to fix by filling `nan` with the unknown label

**Comment by dawe** [comment]
Fixed in 5c0a26edc81cf807b2470af820017716e7568861
